### PR TITLE
GEOPY-4: geoh5 file format docs

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-4 - port geoh5 format description from textile to reST** 
 - Translate existing textile file to rst
- Build base structure for the docs
- Minimal re-structuring of the layout
- Create placeholders for all entity types with uuid.

### CODE REVIEW

**Comment by jincandescent** [comment]
I have attached my comments as a text file to the issue in Jira. I didn't have the time to apply them.

**Review by jincandescent — CHANGES_REQUESTED** [review]
See comments file in Jira issue.

**Comment by jincandescent** [comment]
One more thing - For the referenced data here : https://geoh5py.readthedocs.io/en/latest/content/geoh5_format/analyst/data.html#referenced
...instead of noting just "Value map", we should link to the Data type definition that has the full description of what referenced types need to include

**Comment by jincandescent** [comment]
The sections about the specific GI types, Tools group and other specialized stuff should probably mention that these types have very specific uses and requirements not meant for general use, but their type ID is included here for completeness.

**Comment by domfournier** [comment]
> The sections about the specific GI types, Tools group and other specialized stuff should probably mention that these types have very specific uses and requirements not meant for general use.

Do you think we should move all the integrator stuff to a specific page instead of referencing inside groups and objects sections?

File: docs/content/geoh5_format/analyst/objects.rst [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+
+Each cell represents a point of a 3D rectilinear grid. For a 3D cell
+index (i,j,k) along axes U,V and Z of length nU, nV and nZ respectively,
+
+::
+
+   cell index = k + i*nZ + j*nU*nZ
+
+- Without rotation angles, U points eastwards, V points northwards, and
+   Z points upwards.
```
Comment (benk-mira) [inline docs/content/geoh5_format/analyst/objects.rst:48]: In html the first line is bold and the second indented line is not.  Is this intended?

Comment (benk-mira) [reply docs/content/geoh5_format/analyst/objects.rst:48]: ... it's like this all through the document

File: docs/content/geoh5_format/analyst/objects.rst [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+
+**UUID : {77AC043C-FE8D-4D14-8167-75E300FB835A}**
+
+*Not yet geoh5py implemented*
+
+*To be further documented*
+
+- Vertices represent the four corners of the geolocated image. Note** :
+   Should be arranged as a rectangle currently, since Geoscience ANALYST
+   does not currently support skewed images.
```
Comment (benk-mira) [inline docs/content/geoh5_format/analyst/objects.rst:159]: Weird bold pattern

File: docs/content/geoh5_format/hierarchy.rst [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+
+Groups
+======
+
+.. figure:: ./images/groups.png
+    :align: right
+    :width: 300
+
+Groups are used as container for Objects to organize the Workspace.
```
Comment (benk-mira) [inline docs/content/geoh5_format/hierarchy.rst:54]: *containers

File: docs/content/geoh5_format/hierarchy.rst [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+has a type that defines how its HDF5 datasets should be interpreted**.
+This type is shared among any number of entities (groups/objects/data sets).
+
+.. _group_types:
+
+Group Types
+-----------
+
+Groups are simple container for other groups and objects. It
+is often used to assign special meanings to a collection of entities or to create specialized software
```
Comment (benk-mira) [inline docs/content/geoh5_format/hierarchy.rst:174]: *containers

Comment (benk-mira) [reply docs/content/geoh5_format/hierarchy.rst:174]: *They are often used

**Review by benk-mira — APPROVED** [review]
Looks great!  Just a few minor suggestions, but I'll approve and let you decide what's worth fixing up.

File: docs/content/geoh5_format/analyst/objects.rst [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+   described below, block models do not require a Vertices or Cells
+   dataset.
+
+Datasets
+^^^^^^^^
+- **U cell delimiters** : 1D array of ``double``
+    Distances of cell edges from origin along the U axis (first value should be 0)
+- **V cell delimiters** : 1D array of ``double``
+    Distances of cell edges from origin along the V axis (first value should be 0)
+- **Z cell delimiters** : 1D array of ``double``
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/content/geoh5_format/analyst/objects.rst:59]: I guess cause we can't rotate the z-axis.

File: docs/content/geoh5_format/analyst/objects.rst [hunk 6]
```diff
@@ -0,0 +1,476 @@
...
+
+   cell index = k + i*nZ + j*nU*nZ
+
+- Without rotation angles, U points eastwards, V points northwards, and
+   Z points upwards.
+- Since their geometry is defined entirely by the additional data
+   described below, block models do not require a Vertices or Cells
+   dataset.
+
+Datasets
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/content/geoh5_format/analyst/objects.rst:53]: Change attributes to Field lIst. SHould be clearer

File: docs/content/geoh5_format/analyst/objects.rst [hunk 7]
```diff
@@ -0,0 +1,476 @@
...
+^^^^^^^^
+- **U cell delimiters** : 1D array of ``double``
+    Distances of cell edges from origin along the U axis (first value should be 0)
+- **V cell delimiters** : 1D array of ``double``
+    Distances of cell edges from origin along the V axis (first value should be 0)
+- **Z cell delimiters** : 1D array of ``double``
+    Distances of cell edges from origin upwards along the vertical axis (first value should be 0)
+
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/content/geoh5_format/analyst/objects.rst:61]: Same comment as above

File: docs/content/geoh5_format/analyst/objects.rst (hunk 1 above)
Comment (domfournier) [reply docs/content/geoh5_format/analyst/objects.rst:48]: good catch, will change all to plain text

File: docs/content/geoh5_format/hierarchy.rst (hunk 3 above)
Comment (domfournier) [reply docs/content/geoh5_format/hierarchy.rst:54]: Done

File: docs/content/geoh5_format/hierarchy.rst (hunk 4 above)
Comment (domfournier) [reply docs/content/geoh5_format/hierarchy.rst:174]: DOne

**Pull request merged**